### PR TITLE
add periodic timeout build option to OTA requestor efr32, add software version #define to efr32 applications

### PR DESCRIPTION
**Pull request by mykrupp**
#### Problem 1 
The core OTA functionality is supposed to have the periodic query timeout as a config parameter to the OTARequestor class. This is not currently supported by the OTARequestor (Feb 7 2022) but when it is implemented the EFR32 app needs to pass this parameter to the OTARequestor. 

#### Change overview
This adds an option periodic query timeout variable to gn_efr32_example.sh

#### Testing
I ran this test using a lighting example and added a print statement in the code. I saw that modifying the build option, OTA_periodic_query_timeout, would allow me to change the #define TIMEOUT in the light app code. 

#### Problem 2 - 
The EFR example apps need to implement the Software Version functionality. 
There must be a way to assign a numeric software version to the application image. This could possibly be a compile flag or a #define macro. 
The version value must written to the SoftwareVersion attribute of the Basic cluster. 

#### Change overview
Added #define for the string parameter of the basic cluster

#### Testing
No testing yet because only #define was added

#### Problem 3
light-switch app needs chip-enable-ota-requestor defined

#### Change overview
Added this in the BUILD.gn file and args.gni file

#### Testing 
Ensure that this command succeeds -
commander gbl create --app out/debug/BRD4161A/chip-efr32-light-switch-example.s37 myOtaSwicth_compress.gbl --compress lzma

### CODE REVIEW

File: examples/lighting-app/efr32/src/main.cpp [hunk 1]
```diff
@@ -1,227 +0,0 @@
-/*
```
Comment (jmartinez-silabs) [inline examples/lighting-app/efr32/src/main.cpp:1]: What happened to this? why are we deleting it? I don't see anything that replaces it

File: examples/light-switch-app/efr32/include/AppConfig.h [hunk 2]
```diff
@@ -21,6 +21,9 @@
 
 // ---- Lighting Example App Config ----
 
+// Software Version
+#define SW_VERSION "1.0.0"
```
Comment (jmartinez-silabs) [inline examples/light-switch-app/efr32/include/AppConfig.h:25]: We have a software version #define for all apps already. We should reuse that

`#define CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION_STRING "0.1ALPHA"`

File: examples/lighting-app/efr32/src/main.cpp (hunk 1 above)
Comment (mykrupp) [reply examples/lighting-app/efr32/src/main.cpp]: I was using main to test, but I only mean to delete it from the the PR, not the repo

Comment (selissia) [reply examples/lighting-app/efr32/src/main.cpp]: Well, this file needs to be undeleted somehow :)

File: examples/lock-app/efr32/include/AppConfig.h [hunk 3]
```diff
@@ -21,6 +21,9 @@
 
 // ---- Lock Example App Config ----
 
+// Software Version
+#define SW_VERSION "1.0.0"
```
Comment (selissia) [inline examples/lock-app/efr32/include/AppConfig.h]: Software version needs to be a uint32_t, please see Section 11.1.6 of the Matter spec, attributes of the Basic cluster

Comment (selissia) [reply examples/lock-app/efr32/include/AppConfig.h]: We also need a separate parameter that we'll use to write the SoftwareVersionString attribute of the basic cluster

File: src/app/clusters/ota-requestor/OTARequestor.h [hunk 4]
```diff
@@ -31,6 +31,10 @@
 
 #include "BDXDownloader.h"
 
+#ifdef OTA_PERIODIC_TIMEOUT
```
Comment (selissia) [inline src/app/clusters/ota-requestor/OTARequestor.h:34]: We shouldn't be modifying this file. The change should go in the EFR files and we'll call the OTA API from there (will probably do it in a separate PR)

File: examples/lock-app/efr32/include/AppConfig.h (hunk 3 above)
Comment (mykrupp) [reply examples/lock-app/efr32/include/AppConfig.h]: Added a separate parameter, put it in the correct format and moved it to the CHIPProjectConfig.h files

**Review by jmartinez-silabs — APPROVED** [review]

**Review by selissia — APPROVED** [review]

File: examples/lighting-app/efr32/BUILD.gn [hunk 5]
```diff
@@ -48,6 +48,9 @@ declare_args() {
   # Enable Sleepy end device
   enable_sleepy_device = false
 
+  # OTA timeout
+  OTA_periodic_query_timeout = 10
```
Comment (carol-apple) [inline examples/lighting-app/efr32/BUILD.gn:52]: nit: specify the unit of time here

**Review by carol-apple — APPROVED** [review]
